### PR TITLE
Add tests for network adapter down/up/disconnect/connect/hotremove operations

### PR DESCRIPTION
**Pull request by keirazhang**
Update network_device_ops to test network adapter when:
1. Bring down/up the network adapter in guest, then ping vlan gateway
2. Disconnect/reconnect the network adapter on VM, then ping vlan gateway
3. Hot remove the network adapter, then check the network interface is removed in guest

Signed-off-by: Qi Zhang <qiz@vmware.com>

### CODE REVIEW

**Comment by keirazhang** [comment]
```
+-----------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                  |
|                           | bitness='64'                        |
|                           | distroName='VMware Photon OS'       |
|                           | distroVersion='3.0'                 |
|                           | familyName='Linux'                  |
|                           | kernelVersion='4.19.245-2.ph3-esx'  |
|                           | prettyName='VMware Photon OS/Linux' |
+-----------------------------------------------------------------+


Test Results (Total: 2, Passed: 2, Elapsed Time: 00:06:54)
+-------------------------------------------------+
| Name                       | Status | Exec Time |
+-------------------------------------------------+
| e1000e_network_device_ops  | Passed | 00:03:32  |
| vmxnet3_network_device_ops | Passed | 00:02:29  |
+-------------------------------------------------+

```
```
+-------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                          |
|                           | bitness='64'                                |
|                           | distroName='AlmaLinux'                      |
|                           | distroVersion='9.0'                         |
|                           | familyName='Linux'                          |
|                           | kernelVersion='5.14.0-70.13.1.el9_0.x86_64' |
|                           | prettyName='AlmaLinux 9.0 (Emerald Puma)'   |
+-------------------------------------------------------------------------+


Test Results (Total: 2, Passed: 2, Elapsed Time: 00:07:33)
+-------------------------------------------------+
| Name                       | Status | Exec Time |
+-------------------------------------------------+
| e1000e_network_device_ops  | Passed | 00:04:00  |
| vmxnet3_network_device_ops | Passed | 00:02:39  |
+-------------------------------------------------+

```
```
+-------------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                                |
|                           | bitness='64'                                      |
|                           | distroName='Red Hat Enterprise Linux'             |
|                           | distroVersion='8.6'                               |
|                           | familyName='Linux'                                |
|                           | kernelVersion='4.18.0-372.9.1.el8.x86_64'         |
|                           | prettyName='Red Hat Enterprise Linux 8.6 (Ootpa)' |
+-------------------------------------------------------------------------------+


Test Results (Total: 2, Passed: 2, Elapsed Time: 00:08:50)
+-------------------------------------------------+
| Name                       | Status | Exec Time |
+-------------------------------------------------+
| e1000e_network_device_ops  | Passed | 00:04:21  |
| vmxnet3_network_device_ops | Passed | 00:03:23  |
+-------------------------------------------------+
```
```
+------------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                               |
|                           | bitness='64'                                     |
|                           | distroName='Red Hat Enterprise Linux'            |
|                           | distroVersion='9.0'                              |
|                           | familyName='Linux'                               |
|                           | kernelVersion='5.14.0-70.13.1.el9_0.x86_64'      |
|                           | prettyName='Red Hat Enterprise Linux 9.0 (Plow)' |
+------------------------------------------------------------------------------+


Test Results (Total: 2, Passed: 2, Elapsed Time: 00:08:44)
+-------------------------------------------------+
| Name                       | Status | Exec Time |
+-------------------------------------------------+
| e1000e_network_device_ops  | Passed | 00:04:25  |
| vmxnet3_network_device_ops | Passed | 00:03:18  |
+-------------------------------------------------+
```
```
+---------------------------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                                              |
|                           | bitness='64'                                                    |
|                           | buildNumber='2022-08-29-1855'                                   |
|                           | distroName='Flatcar Container Linux by Kinvolk'                 |
|                           | distroVersion='3227.2.2'                                        |
|                           | familyName='Linux'                                              |
|                           | kernelVersion='5.15.63-flatcar'                                 |
|                           | prettyName='Flatcar Container Linux by Kinvolk 3227.2.2 (Oklo)' |
+---------------------------------------------------------------------------------------------+


Test Results (Total: 2, Passed: 2, Elapsed Time: 00:06:20)
+-------------------------------------------------+
| Name                       | Status | Exec Time |
+-------------------------------------------------+
| e1000e_network_device_ops  | Passed | 00:03:19  |
| vmxnet3_network_device_ops | Passed | 00:02:08  |
+-------------------------------------------------+
```
```
+------------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                               |
|                           | bitness='64'                                     |
|                           | distroName='SLES'                                |
|                           | distroVersion='15.4'                             |
|                           | familyName='Linux'                               |
|                           | kernelVersion='5.14.21-150400.22-default'        |
|                           | prettyName='SUSE Linux Enterprise Server 15 SP4' |
+------------------------------------------------------------------------------+


Test Results (Total: 2, Passed: 2, Elapsed Time: 00:08:20)
+-------------------------------------------------+
| Name                       | Status | Exec Time |
+-------------------------------------------------+
| e1000e_network_device_ops  | Passed | 00:04:12  |
| vmxnet3_network_device_ops | Passed | 00:03:12  |
+-------------------------------------------------+
```
```
+-----------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                  |
|                           | bitness='64'                        |
|                           | distroName='VMware Photon OS'       |
|                           | distroVersion='4.0'                 |
|                           | familyName='Linux'                  |
|                           | kernelVersion='5.10.83-6.ph4-esx'   |
|                           | prettyName='VMware Photon OS/Linux' |
+-----------------------------------------------------------------+


Test Results (Total: 2, Passed: 2, Elapsed Time: 00:06:57)
+-------------------------------------------------+
| Name                       | Status | Exec Time |
+-------------------------------------------------+
| e1000e_network_device_ops  | Passed | 00:03:36  |
| vmxnet3_network_device_ops | Passed | 00:02:24  |
```
```
+----------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                 |
|                           | bitness='64'                       |
|                           | distroName='Ubuntu 22.04 LTS'      |
|                           | distroVersion='22.04'              |
|                           | familyName='Linux'                 |
|                           | kernelVersion='5.15.0-25-generic'  |
|                           | prettyName='Ubuntu 22.04 LTS'      |
+----------------------------------------------------------------+


Test Results (Total: 2, Passed: 2, Elapsed Time: 00:10:02)
+-------------------------------------------------+
| Name                       | Status | Exec Time |
+-------------------------------------------------+
| e1000e_network_device_ops  | Passed | 00:05:01  |
| vmxnet3_network_device_ops | Passed | 00:04:01  |
+-------------------------------------------------+
```
```
+-----------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                        |
|                           | bitness='64'                              |
|                           | distroName='AlmaLinux'                    |
|                           | distroVersion='8.6'                       |
|                           | familyName='Linux'                        |
|                           | kernelVersion='4.18.0-372.9.1.el8.x86_64' |
|                           | prettyName='AlmaLinux 8.6 (Sky Tiger)'    |
+-----------------------------------------------------------------------+


Test Results (Total: 2, Passed: 2, Elapsed Time: 00:08:02)
+-------------------------------------------------+
| Name                       | Status | Exec Time |
+-------------------------------------------------+
| e1000e_network_device_ops  | Passed | 00:04:00  |
| vmxnet3_network_device_ops | Passed | 00:02:59  |
+-------------------------------------------------+
```
```
+---------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                |
|                           | bitness='64'                      |
|                           | distroName='Ubuntu 22.04.1 LTS'   |
|                           | distroVersion='22.04'             |
|                           | familyName='Linux'                |
|                           | kernelVersion='5.15.0-47-generic' |
|                           | prettyName='Ubuntu 22.04.1 LTS'   |
+---------------------------------------------------------------+


Test Results (Total: 2, Passed: 2, Elapsed Time: 00:08:59)
+-------------------------------------------------+
| Name                       | Status | Exec Time |
+-------------------------------------------------+
| e1000e_network_device_ops  | Passed | 00:04:33  |
| vmxnet3_network_device_ops | Passed | 00:03:31  |
+-------------------------------------------------+
```
```
+-------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                          |
|                           | bitness='64'                                |
|                           | distroName='Debian GNU/Linux 11 (bullseye)' |
|                           | distroVersion='11'                          |
|                           | familyName='Linux'                          |
|                           | kernelVersion='5.10.0-16-amd64'             |
|                           | prettyName='Debian GNU/Linux 11 (bullseye)' |
+-------------------------------------------------------------------------+


Test Results (Total: 2, Passed: 2, Elapsed Time: 00:07:46)
+-------------------------------------------------+
| Name                       | Status | Exec Time |
+-------------------------------------------------+
| e1000e_network_device_ops  | Passed | 00:03:52  |
| vmxnet3_network_device_ops | Passed | 00:02:46  |
+-------------------------------------------------+
```

File: linux/network_device_ops/apply_new_network_config.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        dest: "{{ network_conf_path }}"
+        mode: "0666"
+      delegate_to: "{{ vm_guest_ip }}"
+
+    - name: "Print content of network configure file '{{ network_conf_path }}'"
+      ansible.builtin.command: "cat {{ network_conf_path }}"
+      register: network_config
+      changed_when: false
+      delegate_to: "{{ vm_guest_ip }}"
+    - ansible.builtin.debug: var=network_config.stdout_lines
```
Comment (Tomorrow9) [inline linux/network_device_ops/apply_new_network_config.yml:60]: No task name for debug.

File: linux/network_device_ops/apply_new_network_config.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+# Get IP address of new interface
+- name: "Get link '{{ new_network_adapter }}' status"
+  ansible.builtin.shell: "ip -br addr show {{ new_network_adapter }} | awk '{print $3}'"
+  register: link_status
+  changed_when: false
+  until: '"192.168" in link_status.stdout'
+  retries: 20
+  delay: 5
+  delegate_to: "{{ vm_guest_ip }}"
+- ansible.builtin.debug: var=link_status
```
Comment (Tomorrow9) [inline linux/network_device_ops/apply_new_network_config.yml:102]: No task name for debug.

File: linux/network_device_ops/apply_new_network_config.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
+# Copyright 2021-2022 VMware, Inc.
```
Comment (Tomorrow9) [inline linux/network_device_ops/apply_new_network_config.yml:1]: copyright not changed for new files.

**Review by Tomorrow9 — DISMISSED** [review]
LGTM, thanks!

**Review by Tomorrow9 — APPROVED** [review]

**Pull request merged**